vignettes/Data_retrieval_via_API.Rmd

```diff
@@ -0,0 +1,508 @@
+printFormattedTable(asid)
+```
+
+Like the other <font face="CMTT10">tcplLoad</font> functions, <font face="CMTT10">tcplLoadAsid</font> can subset with as many `fld`s as desired and can add any available additional fields.
+
+```{r load_asid_all}
+# subset all asids by using multiple fields -- val must be same length in list form!
+asids <- tcplLoadAsid(fld = c("organism", "tissue"), 
+                      val = list("rat", "liver"),
+                      add.fld = c("aeid", "acid", "asid", "signal_direction")) 
+```
+
+```{r, echo = FALSE}
+printFormattedTable(asids)
+```
+
+### Load unit
+
+To load the normalized data type, or response unit, use <font face="CMTT10">tcplLoadUnit</font>. Notice how <font face="CMTT10">tcplLoadUnit</font> does not use `fld` and `val`, but uses `aeid` as input.
+
+```{r load_unit}
+# load resp_unit for given aeid
+unit <- tcplLoadUnit(aeid = c(663, 891))
+```
+
+```{r, echo = FALSE}
+printFormattedTable(unit)
+```
+
+## Load sample/chemical information
+
+### Load concentration unit
+
+Use <font face="CMTT10">tcplLoadConcUnit</font> to load the concentration unit for a specific `spid` or multiple `spid`s.
+
+```{r load_conc_unit}
+# load conc_unit for given spid
+conc_unit <- tcplLoadConcUnit(spid = "TP0000904H05")
+```
+
+```{r, echo = FALSE}
+printFormattedTable(conc_unit)
+```
+
+### Load chemical info
+
+Use <font face="CMTT10">tcplLoadChem</font> to load the chemical information for a specific `spid` or multiple `spid`s. Notice tcplLoadChem uses `field` instead of `fld`.
+
+```{r load_chem}
+# load chem_info for given spid
+chem_info <- tcplLoadChem(field = "spid", val = "TP0000904H05")
+```
+
+```{r, echo = FALSE}
+printFormattedTable(chem_info)
+```
+
+# Plotting
+
+<font face="CMTT10">tcplPlot</font> is tcpl’s single flexible plotting function, allowing for interactive and consistent visualization of concentration-response curves via customizable parameters. For more details on implementation, parameters and specific customization instructions, refer to the main Data Retrieval vignette. This section will instead focus on the limitations of using the CCTE API as a data source for <font face="CMTT10">tcplPlot</font>.
+
+Like when loading data, the user must indicate the applicable fields and ids which are query-able via the CCTE API. These such fields include and are limited to "AEID", "m4id", "SPID", and "DTXSID". Any other fields will result in error. While supplying multiple ids (through `val`) are valid, multiple fields are not, so combinations of fields are not compatible with <font face="CMTT10">tcplPlot</font> when connected to the API. For example, if `fld = "spid"`, no aeid can be specified, meaning every matching spid will be plotted. If `fld = "aeid"`, every sample within the given endpoint(s) will be plotted. If `fld = "m4id"`, only one plot will output for every each m4id input. Therefore, if looking for a specific aeid/spid combo, one should determine the corresponding m4id, like below.
```

Comment:
Stylistic suggestion: Consider alternative phrasing to "Like when loading data, " in line 430.